### PR TITLE
Merge MatchingEngine Queues

### PR DESCRIPTION
**Pull request by itsbalamurali**
Implements #166

### CODE REVIEW

File: lib/orderbook/MatchingEngine.ts [hunk 1]
```diff
@@ -153,17 +145,13 @@ class MatchingEngine {
     let addTo: PriorityQueue|null = null;
 
     if (isBuyOrder) {
-      matchAgainst.push(this.priorityQueues.peerSellOrders);
-      if (this.internalMatching) {
-        matchAgainst.unshift(this.priorityQueues.ownSellOrders);
-        addTo = this.priorityQueues.ownBuyOrders;
-      }
+      matchAgainst.push(this.priorityQueues.sellOrders);
+      matchAgainst.unshift(this.priorityQueues.sellOrders);
```
Comment (moshababo) [inline lib/orderbook/MatchingEngine.ts]: why adding it to the same queue twice?

File: lib/orderbook/OrderBook.ts [hunk 2]
```diff
@@ -30,25 +30,19 @@ class OrderBook {
...
-    const peerSellOrdersByPairs = groupBy(peerOrders.sellOrders, order => order.pairId);
-    const ownBuyOrdersByPairs = groupBy(ownOrders.buyOrders, order => order.pairId);
-    const ownSellOrdersByPairs = groupBy(ownOrders.sellOrders, order => order.pairId);
+    const buyOrdersByPairs = groupBy(Orders.buyOrders, order => order.pairId);
+    const sellOrdersByPairs = groupBy(Orders.sellOrders, order => order.pairId);
 
     pairs.forEach((pair) => {
       this.matchingEngines[pair.id] = new MatchingEngine(
         pair.id,
-        this.config.internalmatching,
```
Comment (moshababo) [inline lib/orderbook/OrderBook.ts]: remove this config throughout the project as well

**Comment by kilrau** [comment]
As discussed, ready by tomorrow Thursday 12th

**Comment by itsbalamurali** [comment]
@moshababo Please have a look. Tests are failing since we are no longer persisting orders to database post matching partially to return the remaining order quantity.

**Comment by moshababo** [comment]
@itsbalamurali tests should be fixed as well

**Comment by itsbalamurali** [comment]
@moshababo i have temporarily removed the test case as it has been fixed as part of #199 already.

**Comment by itsbalamurali** [comment]
I request you to have a look, when you can.

**Pull request merged**